.circleci/config.yml

```diff
@@ -29,3 +29,85 @@ jobs:
           key: v1-pkg-cache-{{ checksum "go.sum" }}
           paths:
             - "/go/pkg"
+
+  integration:
+    docker:
+      - image: circleci/golang:1.13-stretch
+    resource_class: medium
+    steps:
+      - checkout
+      # The following step are shared with netsec-ethz/rains
+      - run:
+          name: Checkout SCION
+          command: |
+            # Get scion version from go modules. The sed command extracts the
+            # replacement if one is specified in the go.mod file.
+            scion_mod_version=$(go list -m github.com/scionproto/scion | sed -s 's/.*\s*=>\s*//')
+            scion_repo=https://${scion_mod_version% *}.git
+            # Extract the commit hash, which is the last part of the go modules version
+            scion_commit=${scion_mod_version##*-}
+            git clone -b master --single-branch ${scion_repo} ~/scion
```

Comment:
Why master and single-branch? The checkout below will fail when using a commit not on master.

Reply:
Cloning everything was slow. You are right, from the replace directive we might get a commit that is not on the master branch